src/components/templates/PosterHall/components/PosterPreview/PosterPreview.tsx

```diff
@@ -39,13 +74,78 @@ export const PosterPreview: React.FC<PosterPreviewProps> = ({
     [categories]
   );
 
+  const { relatedVenueIds } = useRelatedVenues({
+    //isLoading, relatedVenues,
+    currentVenueId: venueId,
+  });
+
+  const {
+    // isEventsLoading,
+    events: relatedVenueEvents = emptyRelatedEvents,
+  } = useVenueEvents({
+    venueIds: relatedVenueIds,
+  });
+
+  const bookmarkPoster = useCallback(() => {
+    setBookmarkPoster(!isBookmarkedPoster);
+    personalizedPoster.isSaved = !personalizedPoster.isSaved;
+    if (userId && personalizedPoster.id) {
+      savePosterToProfile({
+        venueId: personalizedPoster.id,
+        userId: userId,
+        removeMode: !personalizedPoster.isSaved,
+      });
+    }
+    relatedVenueEvents
+      .filter((event) => event.venueId === personalizedPoster.id)
+      .map((event) => {
+        userId &&
+          event.id &&
+          updatePersonalizedSchedule({
+            event: event,
+            userId: userId,
+            removeMode: !personalizedPoster.isSaved,
+          });
+        return {};
+      });
+    return relatedVenueEvents;
+  }, [userId, isBookmarkedPoster, personalizedPoster, relatedVenueEvents]);
```

Comment:
Whitespace makes things like this much easier to read:

```suggestion
  const bookmarkPoster = useCallback(() => {
    setBookmarkPoster(!isBookmarkedPoster);
    
    personalizedPoster.isSaved = !personalizedPoster.isSaved;
    
    if (userId && personalizedPoster.id) {
      savePosterToProfile({
        venueId: personalizedPoster.id,
        userId: userId,
        removeMode: !personalizedPoster.isSaved,
      });
    }
    
    relatedVenueEvents
      .filter((event) => event.venueId === personalizedPoster.id)
      .map((event) => {
        userId &&
          event.id &&
          updatePersonalizedSchedule({
            event: event,
            userId: userId,
            removeMode: !personalizedPoster.isSaved,
          });
          
        return {};
      });
      
    return relatedVenueEvents;
  }, [userId, isBookmarkedPoster, personalizedPoster, relatedVenueEvents]);
```

Reply:
All this is now pulled into it's own `PosterBookmark.tsx` file